### DISCUSSION TITLE
Web Updater OTA

### DISCUSSION THREAD

**Discussion by Whale4** [Mair/esp32-course#114] [Course Qeustions] [answered]
Hello to everybody! 

There is any chance to implement OTA update from WEB server hosted on the esp32? 

Let say ESP32 run web server, after we connect to this web server via browser and perform upload .bin file for OTA

How it can be done? 
My thoutghs are that it should be uploaded .bin file to esp32 web server to the spiffs partition, and after that esp32 should initiate OTA process from this file

There are any examples how I can do it? (With ESP IDF, not ArduinoIDE) 

There is example of SPIFFS file server on ESP32
https://github.com/espressif/esp-idf/tree/master/examples/protocols/http_server/file_serving

But how then tell to OTA process take .bin file from those SPIFFS place?

**Comment by Mair** [comment, marked as answer]
Hi @Whale4,
It can be done however you would probably need to use a larger storage. what I mean by this is that the typically .bin files fills the entire partition so it will need to be larger then the code base. You can use something like an SDCard and store the .bin file on that. Then all you need to do is serve the file which is not hard.

**Reply by Whale4** [reply]
Thanky you so much!
